src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/DocumentProcessor.java

```diff
@@ -0,0 +1,103 @@
+
+/**
+ * Main component that is run in intervals.
+ *
+ * Looks up all available containers for zip files and extracts pdf files from them.
+ */
+@Component
+public class DocumentProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DocumentProcessor.class);
+
+    private final CloudBlobClient cloudBlobClient;
+    private final Consumer<List<PDF>> pdfsConsumer;
+
+    @Autowired
+    public DocumentProcessor(CloudBlobClient cloudBlobClient, Consumer<List<PDF>> pdfsConsumer) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.pdfsConsumer = pdfsConsumer;
+    }
+
+    @Scheduled(fixedDelayString = "${scan.interval}")
+    public void readBlobs() {
+        cloudBlobClient.listContainers().forEach(cloudBlobContainer -> {
+            try {
+                String containerName = cloudBlobContainer.getName();
+                LOGGER.info("Processing {} container", containerName);
+                CloudBlobContainer container = cloudBlobClient.getContainerReference(containerName);
+                readBlobsFromContainer(container);
+            } catch (URISyntaxException e) {
+                LOGGER.warn("Invalid URL", e);
+                // TODO: track exception in AppInsights Telemetry
+            } catch (StorageException e) {
+                LOGGER.error("Could not obtain container reference", e);
+                // TODO: track exception in AppInsights Telemetry
+            }
+        });
+    }
+
+    private void readBlobsFromContainer(CloudBlobContainer container) {
+        StreamSupport.stream(container.listBlobs().spliterator(), false)
+            .filter(CloudBlob.class::isInstance)
+            .map(CloudBlob.class::cast)
+            .filter(cloudBlob -> cloudBlob.getName().toLowerCase().endsWith(".zip"))
+            // TODO: notify supplier about non-zip files
+            .map(cloudBlob -> {
+                try {
+                    BlobInputStream blobInputStream = container.getBlockBlobReference(cloudBlob.getName())
+                        .openInputStream();
+
+                    return processZipFile(blobInputStream);
+                } catch (StorageException e) {
+                    LOGGER.error("Could not download data", e);
+                    // TODO: track exception in AppInsights Telemetry
+                } catch (URISyntaxException e) {
+                    LOGGER.warn("Invalid URL", e);
+                    // TODO: track exception in AppInsights Telemetry
+                }
+                return null;
+            })
+            .filter(Objects::nonNull)
+            .forEach(pdfsConsumer);
+    }
+
+    private List<PDF> processZipFile(BlobInputStream blobInputStream) {
+        List<PDF> pdfs = new ArrayList<>();
+
+        try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
+            ZipEntry ze;
+            while ((ze = zis.getNextEntry()) != null) {
+                PDF pdf = new PDF(ze.getName(), ByteStreams.toByteArray(zis));
```

Comment:
will the zip file contain only PDFs?

Reply:
In this ticket we are only reading PDFs.